addons/pos_self_order/controllers/orders.py

```diff
@@ -145,18 +162,92 @@ def get_tables(self, access_token):
 
         return tables
 
-    def _process_lines(self, lines, pos_config, pos_order_id):
+
+    @http.route('/kiosk/payment/<int:pos_config_id>', auth='public', type='json', website=True)
+    def pos_self_order_kiosk_payment(self, pos_config_id, order, payment_method_id, access_token):
+        order_dict = self.process_new_order(order, access_token, table_identifier=None)
+
+        if not order_dict.get('id'):
+            raise BadRequest("Something went wrong")
+
+        order_sudo = request.env['pos.order'].sudo().browse(order_dict.get('id'))
+        payment_method_sudo = request.env["pos.payment.method"].sudo().browse(payment_method_id)
```

Comment:
is there some verifications than the payment method is activated in this config?